apis/v1alpha1/httproute_types.go

```diff
@@ -326,6 +343,38 @@ type HTTPHeaderMatch struct {
 	Values map[string]string `json:"values"`
 }
 
+// HTTPQueryParamMatch describes how to select a HTTP route by matching HTTP
+// query parameters.
+type HTTPQueryParamMatch struct {
+	// Type specifies how to match against the value of the query parameter.
+	//
+	// Support: Extended (Exact)
+	//
+	// Support: Custom (RegularExpression, ImplementationSpecific)
+	//
+	// Since RegularExpression QueryParamMatchType has custom conformance,
+	// implementations can support POSIX, PCRE or any other dialects of regular
+	// expressions. Please read the implementation's documentation to determine
+	// the supported dialect.
+	//
+	// HTTP query parameter matching MUST be case-sensitive for both keys and
+	// values.
+	//
+	// +optional
+	// +kubebuilder:default=Exact
+	Type *QueryParamMatchType `json:"type,omitempty"`
+
+	// Values is a map of HTTP query parameters to be matched. It MUST contain
+	// at least one entry.
```

Comment:
Nit: If this must contain at least one entry, should we set the validation minimum?

Reply:
Unfortunately it does not look like it's possible to validate that with annotations. It does seem like something we should catch with the webhook though (for both this and header matching).